### PR TITLE
set tcp_keepalive_time for tidb on aliyun

### PR DESCRIPTION
**Pull request by DanielZhangQD**
<!--
Thank you for contributing to TiDB Operator! Please read TiDB Operator's [CONTRIBUTING](https://github.com/pingcap/tidb-operator/blob/master/CONTRIBUTING.md) document **BEFORE** filing this PR.
-->

### What problem does this PR solve? <!--add and issue link with summary if exists-->
Set tcp_keepalive_time for tidb on Ali ACK, part of https://github.com/pingcap/tidb-operator/issues/880
### What is changed and how does it work?
Set podSecurityContext for TiDB.
In terraform, enable configuration of net.* sysctls in kubelet args.
### Check List <!--REMOVE the items that are not applicable-->

Tests <!-- At least one of them must be included. -->

 - Manual test (add detailed scripts or steps below)
    - Deploy

      Before this PR:
      ```
      Pod:
      [root@248 aliyun]$kubectl  --kubeconfig credentials/kubeconfig -n       my-cluster exec -it my-cluster-tidb-0 -c tidb -- sh
      / # sysctl -a | grep keepalive
      net.ipv4.tcp_keepalive_intvl = 75
      net.ipv4.tcp_keepalive_probes = 9
      net.ipv4.tcp_keepalive_time = 7200
      
      Kubelet:
      [root@z ~]# ps -ef | grep kubelet
      root      5782     1  2 14:01 ?        00:02:58 /usr/bin/kubelet       --bootstrap-kubeconfig=...
      
      MySQL:
      MySQL [(none)]> select sleep(960); select tidb_version(); 
      ERROR 2013 (HY000): Lost connection to MySQL server during query
      ERROR 2006 (HY000): MySQL server has gone away
      No connection. Trying to reconnect...
      Connection id:    9639
      Current database: *** NONE ***
      
      +-------------------------------------------------------------------------------------      -------------------------------------------------------------------------------------      -------------------------------------------------------------------------------------      ------------------------------------------+
      | tidb_version()                                                                                                                                                                                                                                                                                                            |
      +-------------------------------------------------------------------------------------      -------------------------------------------------------------------------------------      -------------------------------------------------------------------------------------      ------------------------------------------+
      | Release Version: v3.0.4
      Git Commit Hash: 694e086e7914a8fc0eb601327edb6bcc31d2c7f2
      Git Branch: HEAD
      UTC Build Time: 2019-10-08 08:13:34
      GoVersion: go version go1.12.10 linux/amd64
      Race Enabled: false
      TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
      Check Table Before Drop: false |
      +-------------------------------------------------------------------------------------      -------------------------------------------------------------------------------------      -------------------------------------------------------------------------------------      ------------------------------------------+
      1 row in set (0.01 sec)
      ```
      After this PR:
      ```
      Pod:
      [root@248 aliyun]$kubectl exec -it my-cluster-tidb-0 -n my-cluster -c tidb       -- sh
      / # sysctl -a | grep keepalive
      sysctl: error reading key 'net.ipv6.conf.all.stable_secret': I/O error
      net.ipv4.tcp_keepalive_intvl = 75
      net.ipv4.tcp_keepalive_probes = 9
      net.ipv4.tcp_keepalive_time = 300
      
      Kubelet:
      [root@x ~]# ps -ef | grep kubelet
      root     27380     1  2 16:27 ?        00:00:07 /usr/bin/kubelet       --bootstrap-kubeconfig=... --allowed-unsafe-sysctls=net.*
      
      MySQL:
      
      MySQL [(none)]> select sleep(960); select tidb_version();
      +------------+
      | sleep(960) |
      +------------+
      |          0 |
      +------------+
      1 row in set (16 min 0.00 sec)
      
      +-------------------------------------------------------------------------------------      -------------------------------------------------------------------------------------      -------------------------------------------------------------------------------------      ------------------------------------------+
      | tidb_version()                                                                                                                                                                                                                                                                                                            |
      +-------------------------------------------------------------------------------------      -------------------------------------------------------------------------------------      -------------------------------------------------------------------------------------      ------------------------------------------+
      | Release Version: v3.0.4
      Git Commit Hash: 694e086e7914a8fc0eb601327edb6bcc31d2c7f2
      Git Branch: HEAD
      UTC Build Time: 2019-10-08 08:13:34
      GoVersion: go version go1.12.10 linux/amd64
      Race Enabled: false
      TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
      Check Table Before Drop: false |
      +-------------------------------------------------------------------------------------      -------------------------------------------------------------------------------------      -------------------------------------------------------------------------------------      ------------------------------------------+
      1 row in set (0.01 sec)
      
      MySQL [(none)]> 

      ```
   - Reboot ECS

      tcp_keepalive_time keeps working.
   - Recover ECS by Auto Scaling Group

      tcp_keepalive_time keeps working on new ECS.
   - Scale out TiDB

      tcp_keepalive_time keeps working on new ECS & TiDB.
   - Scale in TiDB
   
      tcp_keepalive_time keeps working on remaining ECS & TiDB.
   - ACK Upgrade 
      Currently, only two versions during ACK cluster creation, 1.12 & 1.14, but upgrade from 1.12 to 1.14 is not supported yet on ACK.

Code changes

 - Has Terraform scripts change

### Does this PR introduce a user-facing change?:
 <!--
 If no, just write "NONE" in the release-note block below.
 If yes, a release note is required:
 Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
 -->
 ```release-note
Set tcp_keepalive_time to 300s for tidb on aliyun.
 ```

### CODE REVIEW

File: deploy/modules/aliyun/tidb-operator/operator.tf [hunk 1]
```diff
@@ -25,13 +25,18 @@ resource "null_resource" "setup-env" {
...
 kubectl apply -f ${path.module}/manifest/alicloud-disk-storageclass.yaml
 echo '${data.template_file.local-volume-provisioner.rendered}' | kubectl apply -f -
 kubectl patch -n kube-system daemonset flexvolume --type='json' -p='[{"op":"replace", "path": "/spec/template/spec/tolerations", "value":[{"operator": "Exists"}]}]'
 helm init
 until helm ls; do
   echo "Wait tiller ready"
 done
+until kubectl get crd globaljobs.jobs.aliyun.com; do
+  echo "Wait for globaljobs crd ready"
+done
```
Comment (cofyc) [inline deploy/modules/aliyun/tidb-operator/operator.tf]: use `kubectl wait --for=condition=Established ...`, example: https://github.com/pingcap/tidb-operator/pull/997/files#diff-67dacabf080ca4dc8e95d2acf12ee36bR388

- kubectl wait is a simple way to wait for conditions
- crd can only be used when it's established

File: deploy/modules/aliyun/tidb-operator/manifest/global-job-controller.yaml [hunk 2]
```diff
@@ -0,0 +1,87 @@
+apiVersion: apps/v1
```
Comment (aylei) [inline deploy/modules/aliyun/tidb-operator/manifest/global-job-controller.yaml]: I'm hesitant to add this job. Job runs asynchronously, we cannot guarantee the kubelets are properly configured when the terraform operation completes. And introducing an uncontrolled process add instabilities to the script.

Possible alternatives:
- update the `attach_node.sh` just like https://github.com/pingcap/tidb-operator/blob/36f505cd2ef32e4d8f25570f70bdb97dafe7f999/deploy/modules/aliyun/tidb-cluster/templates/user_data.sh.tpl#L11 do, which I don't recommend;
- maintain `attach_node.sh` on ourselves.

How do you think?

File: deploy/modules/aliyun/tidb-operator/templates/kubelet-customize-globaljob.yaml.tpl [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        - mountPath: /alicloud-k8s-host
+          name: kubelet-customize
+      hostNetwork: true
+      hostPID: true
+      restartPolicy: Never
+      volumes:
+      - hostPath:
+          path: /
+          type: Directory
+        name: kubelet-customize
```
Comment (cofyc) [inline deploy/modules/aliyun/tidb-operator/templates/kubelet-customize-globaljob.yaml.tpl]: is this the recommended way in aliyun? can you link official docs here?

File: deploy/modules/aliyun/tidb-cluster/values/default.yaml [hunk 4]
```diff
@@ -12,14 +12,24 @@ tikv:
...
       service.beta.kubernetes.io/alicloud-loadbalancer-address-type: intranet
       service.beta.kubernetes.io/alicloud-loadbalancer-slb-network-type: vpc
+  config: |
+    [log]
+    level = "info"
+    [performance]
+    tcp-keep-alive = true
+  podSecurityContext:
+    sysctls:
+    - name: net.ipv4.tcp_keepalive_time
```
Comment (tennix) [inline deploy/modules/aliyun/tidb-cluster/values/default.yaml]: What is the internal load balancer idle timeout of Aliyun?

**Pull request closed without merging**